src/components/form/CollectionForm.jsx

```diff
@@ -12,25 +12,26 @@ const getDefaultCollectionData = () => {
 };
 
 const CollectionForm = (props) => {
-  const {
-    collectionData = getDefaultCollectionData(),
-    onComplete: completeHandler,
-    categoryOptions = [],
-  } = props;
+  const { collectionData = getDefaultCollectionData(), onComplete: completeHandler, categoryOptions = [] } = props;
 
-  const [name, setName] = useState(collectionData.collectionName);
-  const [description, setDescription] = useState(collectionData.collectionDescription);
+  const [collectionName, setCollectionName] = useState(collectionData.collectionName);
+  const [collectionDescription, setCollectionDescription] = useState(collectionData.collectionDescription);
   const [categoryId, setCategory] = useState(null);
 
-  console.log(categoryOptions);
+  useEffect(() => {
+    if (props.collectionData) {
+      setCollectionName(props.collectionData.collectionName);
+      setCollectionDescription(props.collectionData.collectionDescription);
+      setCategory(props.collectionData.categoryId);
+    }
+  }, [props.collectionData]);
+
```

Comment:
Can we check if this part is still needed i.e. if it works without this useEffect block?

Reply:
It will but there may be unnecessary executions